pkg/controller/operatorsource/operatorsource_controller_test.go

```diff
@@ -0,0 +1,201 @@
+package operatorsource_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	marketplacev1alpha1 "github.com/operator-framework/operator-marketplace/pkg/apis/marketplace/v1alpha1"
+	"github.com/operator-framework/operator-marketplace/pkg/controller/operatorsource"
+	mocks "github.com/operator-framework/operator-marketplace/pkg/mocks/operatorsource_mocks"
+	"github.com/operator-framework/operator-marketplace/pkg/phase"
+	"github.com/stretchr/testify/assert"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// Use Case: If the object has already been cleaned up
+// Expected: No error, no phase change.
+func TestHandle_MissingObject_NoErrorExpected(t *testing.T) {
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+
+	kubeclient := mocks.NewKubeClient(controller)
+	factory := mocks.NewMockPhaseReconcilerFactory(controller)
+	transitioner := mocks.NewPhaseTransitioner(controller)
+	opsrcgenerator := mocks.NewMockOperatorSourceGenerator(controller)
+
+	instance := &marketplacev1alpha1.OperatorSource{}
+	namespacedName := types.NamespacedName{Name: "opsrc-foo", Namespace: "marketplace"}
+	kubeClientErr := k8s_errors.NewNotFound(schema.GroupResource{}, "CatalogSourceConfig not found")
+	kubeclient.EXPECT().Get(context.TODO(), namespacedName, instance).Return(kubeClientErr)
```

Comment:
`sigs.k8s.io/controller-runtime/pkg/client` provides a fake client for writing unit test(s) - `client := NewFakeClient(initObjs...)`. I did not have a chance to use this since the previous version of the sdk had package level functions like `sdk.Create`.

But you can check it out if you want to.

Reply:
Any unit tests moving forward should use the new fake client. Please don't add more code that will need changing.

Reply:
Is there any documentation on fake client? I did see that when I was writing these tests, but I couldn't get a trivial example to work easily so I didn't keep researching.

Reply:
operator-framework/operator-sdk/pull/757

I think this should be done in a separate PR.

Reply:
@aravindhp Okay, so for now I can remove the controller tests and I'll create a story to address it. I think that the catalogsourceconfig controller also needs tests so we should probably do both in the same story.

Reply:
The stories are already present in Jira
https://jira.coreos.com/browse/MKTPLC-105
https://jira.coreos.com/browse/MKTPLC-109